dts/bindings/adc/ti,adc-k3.yaml

```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: Apache-2.0
+
+description: TI K3-family ADC
+
+compatible: "ti,adc-k3"
```

Comment:
I honestly see no pattern in compatible strings added by TI. The GPIO is called `davince`, I2C is called `omap` and the ADC is now called `k3`.

Linux documentation says [1] this ADC is compatible with AM335x and K3. So, I think this is just a general ADC block used in multiple chips. I don't think someone will boot Zephyr on AM335x anymore, but your next generation of chips will most likely use this block again. Probably looks odd when the `k4` architecture uses `ti,adc-k3`.

It doesn't look like Zephyr supports multiple compatible strings per driver. So, I personally would just call it `ti,tscadc`.

1: https://www.kernel.org/doc/Documentation/devicetree/bindings/input/touchscreen/ti-tsc-adc.txt

Reply:
I agree that the compatible strings are indeed confusing, but I don't think adding ti,tscadc is a good idea since that would imply a touchscreen controller as well. I am at a loss here but feel free to suggest more compatible strings, maybe "ti,adc-am335x" or "ti,am335x-adc" for linux compatibility is a good idea?

Reply:
@vaishnavachath any thoughts on that topic?

@natto1784 maybe `ti,k3-adc` is indeed the best name and future platforms just use `compatible = "ti,k3-adc", "ti,k4-adc"` in the device-tree. Similar what happened in Linux. Using `am335` here might indicate there is actually support for that product. We also don't have to copy everything from Linux and could start fresh ... since neither the am335 nor am65 are supported.

It looks like that most compatible use `<vendor>,<ip>-adc`.

Reply:
changed the compat string to `ti,am335x-adc` to match linux